utilities/AWSStorage.mjs

```diff
@@ -33,55 +43,162 @@ export default class AWSStorage extends StorageBridge {
     }
     return Array.from(ids);
   }
-  async get(id,options) {
-    let spec = await super.getSpec(id, options);
-    try {
-      let test = new GetObjectCommand({Bucket: this.connector.profile.aws.s3_bucket, 'Key': spec.path})
-      let response = await this.client.send(test);
-      return response.Body;
-      // response.Body.pipe(res); //TODO: return response
-    } catch (e) {
-      let buffer = await axios(item.url, {responseType: 'arraybuffer'});
-      let image = await sharp(buffer.data);
-      image = await spec.process(image);
-      if (image) {
-        await this.client.send(new PutObjectCommand({
-          Bucket: this.connector.profile.aws.s3_bucket,
-          Key: spec.path,
-          ContentType: "image/png",
-          Body: image
-        }))
-        return Buffer.from(image, 'base64');
-      } else {
-        return this.notFound(req, res);
-      }
+
+  async get(id, thumbnailOptions) {
```

Comment:
There is no purpose to making thumbnail a different case. It is a rendering spec like any other. There should be no fixed image specs in this module. It's not responsible for knowing the purpose of an image rendering. It is responsible only for storing images and retrieving them, modified for the given rendering spec.

Reply:
Recently you mentioned 
Works:
http://localhost:8080/media/image/id/creative/EK/llwclfo7cvrictos.image?scale=60,60,cover
Fails:
http://localhost:8080/media/image/id/creative/EK/llwclfo7cvrictos.image?scale=80,80,cover

The method get takes thumbnail options, it checks existing of thumbnails, if he does not find, he just make it, commit this new thumbnail on db, save it in s3 bucket and finally return it to the client. But in the next time, request will be handled fast). So How should it work definitely ? Could you explain me the purpose of media mixin please ? I have small knowledge of this project. 
P.S I've seen README of media mixin, but i discovered not enough things

Reply:
<img width="1440" alt="Screenshot 2024-10-01 at 13 44 49" src="https://github.com/user-attachments/assets/987fa5ff-b6d7-4017-aa50-0afd74bc5874">

Reply:
As you see here the status works correctly, it's switched from staged to live after uploading files. And i use thumbnails array to be noticed of existing them. I need also thumbnails array in the case deleting image

Reply:
I will update the README today. You are right that it does not provide enough information.

The purpose of the media module is to store images, videos and other large blobs. On retrieval it provides on-demand manipulation of the source object. This saves time preparing an image in dozens of different sizes, encodings or other common variations.

For convenience, it remembers every new variation requested so the rendering is only done once for subsequent requests of the same item and rendering specification.

An item is identified by its unique machine generated ID. If storage is a database, it will save the blob and variations of the blob in a singe record. If storage is a file system it will store the original, the rendered variants and any meta data as separate files with the same root identity. 

It knows nothing about thumbnails, banners, facebook sidebar ads, etc. It only knows how to render media objects according to the spec provided.

...I'll expand on this in the readme.

Reply:
@mespr do you have any updates?

@orozDev confirms that thumbnails are indeed generated once and then stored in S3, which works as expected. This approach allows for optimized image handling, reducing the load on the server.

Additionally, @orozDev confirms that in all subsequent requests, the system correctly retrieves the thumbnails from S3 without attempting to recreate them. Therefore, I believe the logic is fully implemented and functioning as intended.

Reply:
So I want to ask about the presets, In the current bluefire (locally) the presets is empty, so i'm just watching the video where you and Yevgenii were discussing this feature. And as I understand, you don't want to keep commits of thumbnails in the db. And the preset will be enough. So preset will be static thing, that is integrated in the code or each account will have different preset? If each account has the own preset, will it be correct to keep this preset in the db? So I can remove the field thumbnails on media table in the db and I can navigate those thumbnails by the preset, can't I ?

Reply:
<img width="1500" alt="Screenshot 2024-10-05 at 21 15 16" src="https://github.com/user-attachments/assets/1c7bacd4-881a-4406-9015-197dcb20c790">
I got it, it's integrated in the code.